pkg/epp/scheduling/types/types.go

```diff
@@ -29,14 +29,79 @@ type LLMRequest struct {
 	RequestId string
 	// TargetModel is the final target model after traffic split.
 	TargetModel string
-	// Prompt is the prompt that was sent in the request body.
-	Prompt string
+	// Data contains the request-body fields that we parse out as user input.
+	Data *LLMRequestData
 	// Headers is a map of the request headers.
 	Headers map[string]string
 }
 
 func (r *LLMRequest) String() string {
-	return fmt.Sprintf("RequestID: %s, TargetModel: %s, PromptLength: %d, Headers: %v", r.RequestId, r.TargetModel, len(r.Prompt), r.Headers)
+	return fmt.Sprintf("RequestID: %s, TargetModel: %s, RequestData: %s, Headers: %v",
+		r.RequestId, r.TargetModel, r.Data.String(), r.Headers)
+}
+
+// LLMRequestData contains the request-body fields that we parse out as user input,
+// to be used in forming scheduling decisions.
+// An LLMRequestData must contain exactly one of CompletionsRequest or ChatCompletionsRequest.
+type LLMRequestData struct {
+	// CompletionsRequest is the representation of the OpenAI /v1/completions request body.
+	Completions *CompletionsRequest `json:"completions,omitempty"`
+	// ChatCompletionsRequest is the representation of the OpenAI /v1/chat_completions request body.
+	ChatCompletions *ChatCompletionsRequest `json:"chat_completions,omitempty"`
+}
+
+func (r *LLMRequestData) String() string {
```

Comment:
nit: I don't think this is needed given you have String method overrides for individual fields.